lib/elixir/lib/calendar/iso.ex

```diff
@@ -258,4 +460,24 @@ defmodule Calendar.ISO do
       _ -> :error
     end
   end
+
+  def valid_date?(year, month, day)
+  def valid_date?(year, month, day) when month in (1..12) and day in (1..31) do
+    :calendar.valid_date(year, month, day) and year <= 9999
```

Comment:
The check for `year <= 9999` existed in the old code already. Why does this extra check exist?

(Old location [at line 160 in Calendar.ISO](https://github.com/elixir-lang/elixir/blob/a2c45a66dd08eaa5d706fc538cce0c6c2ac8175c/lib/elixir/lib/calendar/iso.ex#L160))